src/transport/raw/MessageHeader.h

```diff
@@ -35,8 +35,9 @@ namespace chip {
 /// Convenience type to make it clear a number represents a node id.
 typedef uint64_t NodeId;
 
-constexpr NodeId kUndefinedNodeId = 0xFFFFFFFFFFFFFFFFll;
-constexpr size_t kMaxTagLen       = 16;
+static constexpr NodeId kUndefinedNodeId = 0ULL;
```

Comment:
is static required?

Reply:
My understanding is that constexpr has internal linkage by default and static is not required. I would drop it to have less  text to parse when reading this file.

Reply:
My understand is static defines the object's lifetime during execution; constexpr specifies that the object should be available during compilation. static and constexpr are independent of each other.

Difference between constexpr and static constexpr is that to use static constexpr you ensure the variable is initialized only once instead of multiple times (with same value).

Another reason is we declares the object in a header file here, when multiple translation units include this header, each .cpp file can gets an independent version of the global variable with the same const value,  I know most modern compiler could optimize the extra variables away because these are const, but put static keyword can force compiler only generate a single instance of it in read-only storage.